### PR TITLE
Homework17

### PR DESCRIPTION
**Pull request by Batmansbitch**
Not sure if this correct? Fell way behind and trying to catch up. Please help

### CODE REVIEW

File: src/test/java/Homework17.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Berichicko) [inline src/test/java/Homework17.java]: **Please delete the empty line!** An Empty Line is a "Code Smell"
The short answer is, _They’re more an annoyance than a problem.They won’t actually break anything—which is why your code continues to run—but they make it awkward to read and change your code, especially if you’re dealing with multiple line breaks per line of real code. They also add just a tiny bit of size to your files, which is nice to reduce._

File: src/test/java/Homework17.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        Thread.sleep(2000);
+
+    }
+
+    public void clickViewAllBtn() throws InterruptedException {
+        WebElement viewAllSearchResult = driver.findElement(By.cssSelector("div.results section.song h1 button"));
+        viewAllSearchResult.click();
+        Thread.sleep(2000);
+    }
+}
```
Comment (Berichicko) [inline src/test/java/Homework17.java]: Please,  **Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
 
**One test - One Assert**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```

File: src/test/java/BaseTest.java [hunk 3]
```diff
@@ -1,12 +1,53 @@
...
+                }
+                public static void clickSubmit() throws InterruptedException {
+        WebElement submitButton = driver.findElement(By.cssSelector("button[type='submit']"));
+        submitButton.click();
+        Thread.sleep(2000);
 
 }
+
+
+    }
```
Comment (Berichicko) [inline src/test/java/BaseTest.java]: How can I beautify code in IntelliJ:

The formatting shortcuts in Intellij IDEA are :
For Windows : Ctrl + Alt + L.
For Ubuntu : Ctrl + Alt + Windows + L.
For Mac : ⌥ (Option) + ⌘ (Command) + L.

image

File: src/test/java/BaseTest.java [hunk 4]
```diff
@@ -1,12 +1,53 @@
...
 public class BaseTest {
+public static WebDriver driver = null;
 
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    public static void launchBrowser() {
+        driver = new ChromeDriver();
```
Comment (Berichicko) [inline src/test/java/BaseTest.java]: After the Chrome 111 update, you can no longer kick off a chromedriver instance unless you add an additional chrome option:
"--remote-allow-origins=*"

 ChromeOptions options = new ChromeOptions();
        options.addArguments("--disable-notifications","--remote-allow-origins=*", "--incognito","--start-maximized");
        driver = new ChromeDriver(options);


Below are the list of available and most commonly used arguments for ChromeOptions class

**start-maximized:** Opens Chrome in maximize mode
**incognito:** Opens Chrome in incognito mode
**headless:** Opens Chrome in headless mode
**disable-extensions:** Disables existing extensions on Chrome browser
**disable-popup-blocking:** Disables pop-ups displayed on Chrome browser
**make-default-browser:** Makes Chrome default browser
**version:** Prints chrome browser version
**disable-infobars:** Prevents Chrome from displaying the notification ‘Chrome is being controlled by automated software

For example

```
  @BeforeMethod
  public static void launchBrowser() {
    ChromeOptions options = new ChromeOptions();
    options.addArguments("--disable-notifications", "--remote-allow-origins=*", "--incognito", "--start-maximized");
    driver = new ChromeDriver(options);
    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
  }
```

image

File: src/test/java/Homework17.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (Berichicko) [inline src/test/java/Homework17.java]: Please, add Homework17 in TestNG.xml to run tests

For example

image

Comment (Batmansbitch) [reply src/test/java/Homework17.java]: Thank you so much for taking the time to give me all this useful information. I really do appreciate it.

**Review by Batmansbitch — COMMENTED** [review]
Appreciate all the useful information.

File: src/test/java/Homework17.java (hunk 1 above)
Comment (Batmansbitch) [reply src/test/java/Homework17.java]: Will Do!!!!!

**Pull request merged**